libsmtutil/SMTLib2Interface.cpp

```diff
@@ -212,33 +213,40 @@ std::string SMTLib2Interface::toSExpr(Expression const& _expr)
 	return sexpr;
 }
 
-std::string SMTLib2Interface::toSmtLibSort(Sort const& _sort)
+std::string SMTLib2Interface::toSmtLibSort(solidity::smtutil::SortPointer _sort)
 {
-	switch (_sort.kind)
+	if (!m_sortNames.count(_sort))
```

Comment:
Keeping this seems like an optimization, do we really need it?

Reply:
We need to remember declared custom Sorts in order to be able to parse back the solver response properly.
This was present in `CHCSmtLib2Interface` before, I just moved it to `SMTLib2Interface` to capture all the declared sorts.